cmd/microcloud/test_input.go

```diff
@@ -0,0 +1,59 @@
+//go:build test
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/canonical/microcloud/microcloud/cmd/tui"
+)
+
+func setupAsker(ctx context.Context) (*tui.InputHandler, error) {
+	noColor := os.Getenv("NO_COLOR")
+	if noColor != "" {
+		tui.DisableColors()
+	}
+
+	useTestConsole := os.Getenv("TEST_CONSOLE")
+	if useTestConsole != "1" {
+		return tui.NewInputHandler(os.Stdin, os.Stdout), nil
+	}
+
+	fmt.Fprintf(os.Stderr, "%s\n\n", `
+  Detected 'TEST_CONSOLE=1', MicroCloud CLI is in testing mode. Terminal interactivity is disabled.
+
+  Interactive microcloud commands will read text instructions by line:
+
+cat << EOF | microcloud init
+table:select                # selects an element in the table
+table:select-all            # selects all elements in the table
+table:select-none           # de-selects all elements in the table
+table:up                    # move up in the table
+table:down                  # move down in the table
+table:wait <time.Duration>  # waits before the next instruction
+table:expect <count>        # waits until exactly <count> peers are available, and errors out if more are found
+table:filter <text>         # applies filtering text to the table output
+table:done                  # confirms the table selection and exits the table
+
+# anything not prefixed with 'table:' will be treated as a raw string. This is used for text entry for individual questions
+EOF`)
+
+	// Due to MicroCloud living inside a snap, the binary must output its text to a file in the snap environment.
+	// This file is used for debugging what the CLI sees for each question.
+	filePath := filepath.Join(os.Getenv("SNAP_COMMON"), "test-output")
+	_ = os.Remove(filePath)
+	out, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0777)
```

Comment:
The reason I avoided replacing `stdout` is because of the presence of `fmt.Print` calls throughout the code. These print to `stdout` by default, so we would have to make sure to always use `fmt.Fprint` instead.

Using `fd3` is a good idea though!

Reply:
I've used `fd3` now, and the output is sent to a `debug` file.

Reply:
Great!